esp/platform/tokenserialization.hpp

```diff
@@ -51,4 +60,390 @@ class TokenSerializer
     }
 };
 
+enum DeserializationResult
+{
+    Deserialization_UNKNOWN = -1,  // no deserialization attempted
+    Deserialization_SUCCESS,
+    Deserialization_BAD_TYPE,      // receiving value cannot be const
+    Deserialization_UNSUPPORTED,   // receiving value type not handled
+    Deserialization_INVALID_TOKEN, // token cannot be NULL, empty, or all whitespace
+    Deserialization_NOT_A_NUMBER,  // non-numeric characters found in numeric conversion
+    Deserialization_OVERFLOW,      // number too large to be represented by receiving value
+    Deserialization_UNDERFLOW,     // number too small to be represented by receiving value
+};
+
+class TokenDeserializer
+{
+public:
+    // Convert the contents of buffer from text to the requested numeric type.
+    // The conversion fails if:
+    //  - value is a const type
+    //  - value is not a numeric type
+    //  - buffer is NULL, empty or entirely whitespace characters
+    //  - buffer contains any character not valid for the receiving type
+    //  - buffer contains a number too large for the receiving type
+    //  - buffer contains a number too small for the receiving type
+    template <typename TValue>
+    DeserializationResult deserialize(const char* buffer, TValue& value) const
+    {
+        DeserializationResult result = Deserialization_UNKNOWN;
+
+        if (std::is_const<TValue>())
+        {
+            result = Deserialization_BAD_TYPE;
+        }
+        else if (std::is_arithmetic<TValue>())
+        {
+            const char* ptr = buffer;
+
+            skipWhitespace(ptr);
+            if (!*ptr)
+            {
+                result = Deserialization_INVALID_TOKEN;
+            }
+            else
+            {
+                if (std::is_integral<TValue>())
+                {
+                    if (std::is_signed<TValue>())
```

Comment:
You are correct. The Windows-specific pragmas (all four) fixed the issue we discussed.